views/components/ConfigurationsList/index.jsx

```diff
@@ -415,268 +264,69 @@ const ConfigurationsList = ({ user, classes, theme }) => {
-                      iconStyle={{ color: 'rgba(0, 0, 0, 0.54)' }}
-                      tooltipPosition="top-center"
-                      tooltip="Duplicate"
-                      onClick={() => copyConfig(configs[item])}
-                    >
-                      <Copy />
-                    </IconButton>
-                  )}
-                </div>
-                <FormControlLabel
-                  control={
-                    <Switch
-                      style={{
-                        width: 'auto',
-                      }}
-                      labelStyle={{ color: 'rgba(0, 0, 0, 0.54)' }}
-                      checked={
-                        'config' in preference
-                          ? configs[item]['_id'] == preference['config']
-                          : false
-                      }
-                      onChange={(e, isInputChecked) =>
-                        changeDefaultConfig(e, isInputChecked, item)
-                      }
-                    />
-                  }
-                  label="Default"
-                />
-              </div>
-            </CardActions>
-          </Card>
-        )
-      }
-    }
-    return cards
-  }
-  const shareWithUsers = () => {
-    const { _id } = state.selectedConfig
+  const shareWithUsers = async () => {
+    const { _id } = sharedWithState.selectedConfig
     const configAttributes = {
-      readers: state.shared.map((sharedWith) => sharedWith.value),
+      readers: sharedWithState.shared.map((sharedWith) => sharedWith.value),
     }
-
-    return window
-      .fetch(apiRoutes.configurations.userConfiguration(user.uid, _id), {
-        method: 'PUT',
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        credentials: 'same-origin',
-        body: JSON.stringify(configAttributes),
-      })
-      .then((response) => {
-        if (response.status === 200) {
-          setState((prevState) => {
+    const response = await UserConfigurationsModel.update(
+      uid,
+      _id,
+      configAttributes
+    )
+    if (response.status === 200) {
+      setConfigurations(
+        configurations.map((config) => {
+          if (config._id === _id) {
             return {
-              ...prevState,
-              configurations: update(state.configurations, {
-                [state.selectedConfig['index']]: {
-                  ['readers']: {
-                    $set: state.shared.map((o) => {
-                      return o.value
-                    }),
-                  },
-                },
-              }),
+              ...config,
+              readers: configAttributes.readers,
             }
-          })
-        }
-        closeSearchUsers()
-      })
+          } else return config
+        })
+      )
+    }
+    closeSearchUsers()
   }
+
   const handleChange = (name) => (value) => {
-    let uid = user.uid
-    let names = value.map((o) => {
+    const names = value.map((o) => {
```

Comment:
how do we know `value` is going to be an array field?

Reply:
react-select does this under the hood
From their docs.
type OptionType = { [string]: any }
type OptionsType = Array<OptionType>

type GroupType = {
  [string]: any, // group label
  options: OptionsType,
}

type ValueType = OptionType | OptionsType | null | void

Reply:
this is saying it _might_ be an array but it would also be an `OptionType` or `null` or `void`